### PR TITLE
Fix clippy errors and warnings

### LINKED ISSUE #2: Fix clippy warnings and errors
Fix all clippy warnings and errors (> 90).

**Solution**: I can make a PR to solve this, if accepted

### PR DESCRIPTION
**Pull request by Jerboas86**
Now clippy reports: 
- 0 errors
- 0 warnings

Errors were mainly due to incorrect comparison between floats.

cargo test runs without failures

Fix #2

### CODE REVIEW

**Comment by orottier** [comment]
Thanks! But the tests are really hard too read now. Maybe use something like https://crates.io/crates/float-cmp ?

**Comment by Jerboas86** [comment]
I agree. I will try with float-cmp

**Comment by Jerboas86** [comment]
What do you think about this assertion (with [float_eq crate](https://crates.io/crates/float_eq))?

```rust
use std::convert::TryInto;
use float_eq::assert_float_eq

let channel_data: [f32; 128] = buffer.channel_data(0)[..]
    .try_into()
    .expect("Slice to array conversion failed");

assert_float_eq!(channel_data, [3.; 128], ulps_all <= 1);
```

Instead of: 

```rust
assert!(buffer
    .channel_data(0)
    .iter()
    .zip(&[3.;128])
    .all(|(a, b)| (a - b).abs() < f32::EPSILON));
```

**Comment by orottier** [comment]
Looks very good 👍

**Comment by Jerboas86** [comment]
- The float comp assertions are now more readable thanks to  [float_eq crate](https://crates.io/crates/float_eq)
- Fix some false negative clippy float comparison in `spatial.rs` and `params.rs`

File: tests/media.rs [hunk 1]
```diff
@@ -64,37 +64,72 @@ fn test_media_buffering() {
 
     // should be silent since the media stream did not yield any output
     let output = context.start_rendering();
-    assert_eq!(output.channel_data(0).as_slice(), &[0.; LENGTH]);
+    assert!(output
+        .channel_data(0)
+        .as_slice()
+        .iter()
+        .zip(&[0.; LENGTH])
+        .all(|(a, b)| (a - b).abs() < f32::EPSILON));
```
Comment (orottier) [inline tests/media.rs:72]: This should still be converted to float_eq!

File: tests/offline.rs [hunk 2]
```diff
@@ -29,8 +29,18 @@ fn test_offline_render() {
     assert_eq!(output.number_of_channels(), 2);
     assert_eq!(output.sample_len(), LENGTH);
 
-    assert_eq!(output.channel_data(0).as_slice(), &[-2.; LENGTH]);
-    assert_eq!(output.channel_data(1).as_slice(), &[-2.; LENGTH]);
+    assert!(output
```
Comment (orottier) [inline tests/offline.rs:32]: Still needs float_eq

**Comment by orottier** [comment]
Please have a look at the two leftover files with float_eq, and the merge conflict. Rest looks good so ready to merge after your fixes

File: tests/media.rs (hunk 1 above)
Comment (orottier) [reply tests/media.rs:73]: this is still the case

File: tests/media.rs [hunk 3]
```diff
@@ -64,37 +65,72 @@ fn test_media_buffering() {
...
```
Comment (orottier) [inline tests/media.rs:94]: here

File: tests/offline.rs [hunk 4]
```diff
@@ -142,8 +151,8 @@ fn test_listener() {
     let _ = context.start_rendering();
 
     let listener = context.listener();
-    assert_eq!(listener.position_y().value(), 2.);
-    assert_eq!(listener.position_x().value(), 1.);
+    assert!((listener.position_y().value() - 2.).abs() < f32::EPSILON);
+    assert!((listener.position_x().value() - 1.).abs() < f32::EPSILON);
```
Comment (orottier) [inline tests/offline.rs:155]: can use assert_float_eq

File: tests/offline.rs [hunk 5]
```diff
@@ -172,8 +181,10 @@ fn test_cycle() {
...
-    assert_eq!(
-        output.channel_data(0).as_slice(),
-        &[2.; BUFFER_SIZE as usize]
-    );
+    assert!(output
+        .channel_data(0)
+        .as_slice()
+        .iter()
+        .zip(&[2.; BUFFER_SIZE as usize])
+        .all(|(a, b)| (a - b).abs() < f32::EPSILON),);
```
Comment (orottier) [inline tests/offline.rs:189]: here

**Comment by Jerboas86** [comment]
Sorry for the careless mistakes. I will do better for next PRs. 
They should be all fix now.

**Comment by orottier** [comment]
No worries, the end result is good! I've taken the liberty to make a squash merge out of this to keep the history clean

**Pull request merged**